sources/Inputs/src/AssemblingDataBase.cpp

```diff
@@ -79,6 +80,14 @@ AssemblingDataBase::AssemblingDataBase(const boost::filesystem::path& assembling
   } catch (const xml::sax::parser::ParserException& e) {
     throw Error(DynModelFileReadError, assemblingFilePath.c_str(), e.what());
   }
+
+  std::map<std::string, inputs::AssemblingDataBase::DynamicAutomaton>::const_iterator dynamicAutomatonsIt;
```

Comment:
Iterators should always be declared within the for loop

Reply:
Do we really need to loop afterward? maybe we could get this information on the fly inside the parser?

Reply:
I'm not sure it's possible because xml::sax::parser::Parser is in libxml, a dynawo 3rdParty.

Reply:
I meant in AssemblingXmlDocument